# PHF21A-E16-KO-RNAseq-Cut-Run

**Abstract:**

How cell-type-specific chromatin landscapes emerge and progress during metazoan
ontogenesis remains an important question. Transcription factors are expressed in a cell
type-specific manner and recruit chromatin-regulatory machinery to specific genomic loci.
In contrast, most chromatin-regulatory proteins are expressed broadly and are assumed to
exert the same intrinsic function across cell types. However, human genetics studies have
revealed an unexpected vulnerability of neurodevelopment to chromatin factor mutations
with unknown mechanisms. Here, we report that 13 chromatin regulators undergo
evolutionary-conserved neuron-specific splicing events involving microexons. Two of these
13 factors are integral components of a histone H3K4 demethylase complex; the catalytic
subunit LSD1 and an H3K4me0-reader protein PHF21A. We found that canonical PHF21A
(PHF21A-c) binds to DNA by AT-hook motif, and the neuronal counterpart PHF21A-n lacks
this DNA-binding function. PHF21A-n showed significantly weaker affinity to the histone
H3 tail compared to PHF21A-c.Â Furthermore, neuronal LSD1 splicing led to reduced affinity
of LSD1 to the nucleosome. In-vitro reconstitution of the canonical and neuronal PHF21A-
LSD1 complexes, combined with in-vivo methylation mapping, identified the neuronal
complex as a hypomorphic H3K4 demethylating machinery with reduced nucleosome
engagement. The neuronal PHF21A, albeit its weaker nucleosome binding, is necessary for
normal gene expression and H3K4 landscape in developing neurons. Thus, ubiquitously
expressed chromatin regulatory complexes can exert neuron-specific functions via
alternative splicing of their subunits.
